vector-im/element-android
# [Location tracker] - Fix some location tracking issues (PSF-1000)

## Type of change


## Content


- Use the latest known location among the ones from all providers when starting tracking.
- Adding minimal period of update to notify callback of location change to avoid too many notifications.
- Prioritize locations received from the providers in the following order: “Fused”, “GPS”, other providers
- Avoid calling callback for provider not available when only one of them is disabled (should be called only when all providers are disabled)


## Motivation and context


Closes #5913 

## Screenshots / GIFs





## Tests



- Check location sharing is enabled in the settings
- Go to a room
- Start to share location
- Check the current location is good on the map
- Check the logs of the LocationTracker

## Tested devices

- OS version(s): Android 11

## Checklist
